src/Form/ReactionRuleAddForm.php

```diff
@@ -71,9 +72,18 @@ public function form(array $form, FormStateInterface $form_state) {
       '#title' => $this->t('React on event'),
       '#options' => $options,
       '#required' => TRUE,
+      '#ajax' => $this->getDefaultAjax(),
       '#description' => $this->t('Whenever the event occurs, rule evaluation is triggered.'),
     ];
 
+    $form['event_configuration'] = array();
+    if ($values = $form_state->getValue('events')) {
```

Comment:
This should be read from the entity. You might need to make sure the entity in form state gets updated on ajax submission.